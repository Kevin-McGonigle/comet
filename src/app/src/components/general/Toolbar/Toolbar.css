.tabMenu {
    color: white;
    display: flex;
    flex-direction: row;
    align-items: center;
    flex-grow: 1;
    justify-content: space-between;
}

.menu {
    margin-right: 12px;
}