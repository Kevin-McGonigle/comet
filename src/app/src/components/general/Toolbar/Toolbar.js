import React from 'react';
import styles from './Toolbar.css';
import classnames from 'classnames';
import {Pane, Tab, Tablist} from 'evergreen-ui';
import ToolbarMenu from '../Menu/Menu';

const cx = args => classnames(styles, args);

const Toolbar = props => {
    const {tabState, setTabState, tabMapping} = props;

    const onClickHandler = (event) => {
        const tabTitle = event.currentTarget.id;
        const tabIndex = tabState.tabs.indexOf(tabMapping[tabTitle]);
        setTabState({
            ...tabState,
            selectedIndex: tabIndex,
        });
    }

    return (
        <div className={cx("tabMenu")}>
            <Pane marginTop={12} marginBottom={12} marginLeft={300}>
                <Tablist>
                    {tabState.tabs.map((tab, index) => (
                        <Tab
                            key={tab}
                            id={tab.split(" ").join('')}
                            isSelected={index === tabState.selectedIndex}
                            aria-controls={`panel-${tab}`}
                            color="white"
                            onClick={event => onClickHandler(event)}
                        >
                            {tab}
                        </Tab>
                    ))}
                </Tablist>
            </Pane>

            <div className={cx("menu")}><ToolbarMenu/></div>
        </div>
    )
};

export default Toolbar;