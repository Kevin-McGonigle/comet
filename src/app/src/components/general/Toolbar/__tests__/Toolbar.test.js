import React from 'react';
import Toolbar from '../Toolbar';
import { shallow, mount } from 'enzyme';
import "../../../../setupTests";

test('should render Toolbar succesfully', () => { 
    const tabState= {
        selectedIndex: 0,
        tabs: ["Metrics", "Inheritance Tree", "Abstract Syntax Tree", "Control Flow Diagram", "Dependency Graph"],
        tabContent: []
    };
    const setTabState = jest.fn();
    const component = shallow(<Toolbar tabState={tabState} setTabState={setTabState}/>);
    expect(component.html()).toMatchSnapshot();
});


test('should change to new tab on tab click succesfully', () => { 
    const tabState = {
        selectedIndex: 0,
        tabs: ["Tab", "Tab2"],
        tabContent: []
    };
    const tabMapping = {
        "Tab": "Tab",
        "Tab2": "Tab 2",
    }

    const setTabState = jest.fn();
    const component = mount(<Toolbar tabMapping={tabMapping} tabState={tabState} setTabState={setTabState}/>);
    const tabButton = component.find("#Tab2").first();
    tabButton.simulate('click');
    expect(setTabState).toHaveBeenCalled();
    expect(tabState.selectedIndex == 1);

});
