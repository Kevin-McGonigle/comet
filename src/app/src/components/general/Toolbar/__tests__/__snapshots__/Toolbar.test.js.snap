// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Toolbar succesfully 1`] = `"<div class=\\"tabMenu\\"><div class=\\"css-nil ðŸ“¦mt_12px ðŸ“¦mb_12px ðŸ“¦ml_300px ðŸ“¦box-szg_border-box\\"><div role=\\"tablist\\" class=\\"ðŸ“¦box-szg_border-box\\"><span class=\\"css-11bc0q4 ðŸ“¦color_white ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_28px ðŸ“¦ltr-spc_0 ðŸ“¦h_28px ðŸ“¦dspl_inline-flex ðŸ“¦pl_8px ðŸ“¦pr_8px ðŸ“¦ml_4px ðŸ“¦mr_4px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦txt-deco_none ðŸ“¦box-szg_border-box\\" tabindex=\\"0\\" id=\\"Metrics\\" aria-controls=\\"panel-Metrics\\" aria-selected=\\"true\\" role=\\"tab\\">Metrics</span><span class=\\"css-11bc0q4 ðŸ“¦color_white ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_28px ðŸ“¦ltr-spc_0 ðŸ“¦h_28px ðŸ“¦dspl_inline-flex ðŸ“¦pl_8px ðŸ“¦pr_8px ðŸ“¦ml_4px ðŸ“¦mr_4px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦txt-deco_none ðŸ“¦box-szg_border-box\\" tabindex=\\"0\\" id=\\"Metric Graph\\" aria-controls=\\"panel-Metric Graph\\" aria-selected=\\"false\\" role=\\"tab\\">Metric Graph</span><span class=\\"css-11bc0q4 ðŸ“¦color_white ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_28px ðŸ“¦ltr-spc_0 ðŸ“¦h_28px ðŸ“¦dspl_inline-flex ðŸ“¦pl_8px ðŸ“¦pr_8px ðŸ“¦ml_4px ðŸ“¦mr_4px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦txt-deco_none ðŸ“¦box-szg_border-box\\" tabindex=\\"0\\" id=\\"Tree\\" aria-controls=\\"panel-Tree\\" aria-selected=\\"false\\" role=\\"tab\\">Tree</span></div></div><div class=\\"menu\\"><button class=\\"css-69cngj ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_32px ðŸ“¦ltr-spc_0 ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_16px ðŸ“¦pl_16px ðŸ“¦ml_0px ðŸ“¦mr_16px ðŸ“¦mb_0px ðŸ“¦h_32px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" role=\\"button\\" aria-haspopup=\\"true\\">Menu</button></div></div>"`;
